core/websocket.go

```diff
@@ -25,6 +25,7 @@ func queryWebSocket(endpoint *Endpoint, result *Result) {
 		result.AddError("Error dialing WS:" + err.Error())
 		return
 	}
+	defer ws.Close()
```

Comment:
@heitorPB If the "DialConfig()" fails with an error, you will return from this function without closing the websocket.
Also, 2 lines below here, you are using ws.Write() after the ws.Close was already called. 
Did you run tests checking that?

Reply:
Hey @Jacoblava, The defer statement in Go does not immediately execute the function it precedes. Instead, it pushes the function call onto a stack and only executes it when the surrounding function is about to return.

If the DialConfig() call fails with an error, then the return statement will be executed and defer ws.Close() statement will not be reached or executed, but it also doesn't need to be, because when DialConfig() fails, there is no open websocket connection to close

i would say that the defer is in the good position